File: 30 days of code/Day 17 - More exceptions/More exceptions.cs
/**
 *              MORE EXCEPTIONS | C#
 * Domain       30 DAYS OF CODE
 * Sub domain   DAY 17
 */

using System;
using System.IO;

class Program
{
    //static TextReader read = Console.In;
    static StreamReader read = new StreamReader("input.txt");

    static void Main(string[] args)
    {
        Calculator calculator = new Calculator();
        int queries = Int32.Parse(read.ReadLine());
        while (queries-- > 0)
        {
            string[] line = read.ReadLine().Split(' ');
            int n = int.Parse(line[0]);
            int p = int.Parse(line[1]);
            try
            {
                int answer = calculator.Power(n, p);
                Console.WriteLine(answer);
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
        }
        read.Close();
    }
}

class Calculator
{
    public int Power(int n, int p)
    {
        if (n < 0 || p < 0)
        {
            throw new Exception("n and p should be non-negative");
        }
        return (int)Math.Pow(n, p);
    }
}

File: 30 days of code/Day 17 - More exceptions/More exceptions.java
/**
 *              MORE EXCEPTIONS | JAVA
 * Domain       30 DAYS OF CODE
 * Sub domain   DAY 17
 */

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) throws FileNotFoundException {
        File file = new File("input.txt");
        Scanner scan = new Scanner(file);
        {
            Calculator calculator = new Calculator();
            int testCases = scan.nextInt();
            while (testCases-- > 0) {
                int n = scan.nextInt();
                int p = scan.nextInt();
                try {
                    int answer = calculator.power(n, p);
                    System.out.println(answer);
                } catch (Exception e) {
                    System.out.println(e.getMessage());
                }
            }
        }
        scan.close();
    }
}

class Calculator {
    int power(int n, int p) throws Exception {
        if (n < 0 || p < 0) {
            throw new Exception("n and p should be non-negative");
        }
        return (int)Math.pow(n, p);
    }
}
